src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/DelegateTests.cs

```diff
@@ -0,0 +1,214 @@
+            Assert.Equal(70, HelperMarshal._functionActionResultValueOfAction);
+        }
+
+        [Fact]
+        public static void MarshalActionIntInt()
+        {
+            HelperMarshal._actionResultValue = 0;
+
+            Runtime.InvokeJS(@"
+                var actionDelegate = App.call_test_method (""CreateActionDelegate"", [  ]);
+                actionDelegate(30,40);
+            ");
+
+            Assert.Equal(70, HelperMarshal._actionResultValue);
+        }
+
+        [Fact]
+        public static void MarshalActionFloatIntToIntInt()
+        {
+            HelperMarshal._actionResultValue = 0;
+            Runtime.InvokeJS(@"
+                var actionDelegate = App.call_test_method (""CreateActionDelegate"", [  ]);
+                actionDelegate(3.14,40);
+            ");
+
+            Assert.Equal(43, HelperMarshal._actionResultValue);
+        }
+
+        [Fact]
+        public static void MarshalDelegateMethod()
+        {
+            HelperMarshal._delMethodResultValue = string.Empty;
+            Runtime.InvokeJS(@"
+                var del = App.call_test_method (""CreateDelegateMethod"", [  ]);
+                del(""Hic sunt dracones"");
+            ");
+
+            Assert.Equal("Hic sunt dracones", HelperMarshal._delMethodResultValue);
+        }
+
+        [Fact]
+        public static void MarshalAnonDelegateMethod()
+        {
+            HelperMarshal._delAnonMethodStringResultValue = string.Empty;
+            Runtime.InvokeJS(@"
+                var del = App.call_test_method (""CreateDelegateAnonMethodReturnString"", [  ]);
+                del(""Hic sunt dracones"");
+            ");
+
+            Assert.Equal("Notification received for: Hic sunt dracones", HelperMarshal._delAnonMethodStringResultValue);
+        }
+
+
+        [Fact]
+        public static void MarshalLambdaDelegateMethod()
+        {
+            HelperMarshal._delLambdaMethodStringResultValue = string.Empty;
+            Runtime.InvokeJS(@"
+                var del = App.call_test_method (""CreateDelegateLambdaMethodReturnString"", [  ]);
+                del(""Hic sunt dracones"");
+            ");
+
+            Assert.Equal("Notification received for: Hic sunt dracones", HelperMarshal._delLambdaMethodStringResultValue);
+        }
+
+        [Fact]
+        public static void MarshalDelegateMethodReturnString()
+        {
+            HelperMarshal._delMethodStringResultValue = string.Empty;
+            Runtime.InvokeJS(@"
+                var del = App.call_test_method (""CreateDelegateMethodReturnString"", [  ]);
+                var res = del(""Hic sunt dracones"");
+                App.call_test_method (""SetTestString1"", [ res ]);
+            ");
+
+            Assert.Equal("Received: Hic sunt dracones", HelperMarshal._delMethodStringResultValue);
+        }
+
+        [Fact]
+        public static void MarshalCustomMultiDelegateAcceptingString()
```

Comment:
The multicast delegate ones can be combined too

Reply:
addressed in recent commits